### PR TITLE
snapshot: impose retry logic

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Impose retry logic to snapshotting.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kwokctl/snapshot/save.go [hunk 1]
```diff
@@ -62,23 +65,31 @@ func Save(ctx context.Context, kubeconfigPath string, w io.Writer, resources []s
...
+			listPager := pager.New(func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
+				return nri.List(ctx, opts)
+			})
+			return listPager.EachListItem(ctx, metav1.ListOptions{}, func(obj runtime.Object) error {
+				if o, ok := obj.(metav1.Object); ok {
+					clearUnstructured(o)
+				}
+				return encoder.Encode(obj)
+			})
+		}); err != nil {
```
Comment (wzshiming) [inline pkg/kwokctl/snapshot/save.go:80]: We could have just added retries to the pagination list, which might have worked better.

```suggestion
		listPager := pager.New(func(ctx context.Context, opts metav1.ListOptions) (list runtime.Object, err error) {
			err = retry.OnError(retry.DefaultBackoff, retriable, func() error {
				list, err = nri.List(ctx, opts)
				return err
			})
			return list, err
		})
		err = listPager.EachListItem(ctx, metav1.ListOptions{}, func(obj runtime.Object) error {
			if o, ok := obj.(metav1.Object); ok {
				clearUnstructured(o)
			}
			return encoder.Encode(obj)
		})
		if err != nil {
```

Comment (Huang-Wei) [reply pkg/kwokctl/snapshot/save.go]: +1, we should catch and retry the underneath API call, in which case is the [pageFn](https://github.com/kubernetes/kubernetes/blob/ad18954259eae3db51bac2274ed4ca7304b923c4/staging/src/k8s.io/client-go/tools/pager/pager.go#L237) in the `pager` object.

Updated.

**Pull request merged**